components/tidb_query_expr/src/impl_string.rs

```diff
@@ -499,6 +499,22 @@ pub fn upper(arg: BytesRef, writer: BytesWriter) -> Result<BytesGuard> {
     Ok(writer.write_ref(Some(arg)))
 }
 
+#[rpn_fn(writer)]
+#[inline]
+pub fn lower(arg: BytesRef, writer: BytesWriter) -> Result<BytesGuard> {
```

Comment:
ping @andylokandy , Do we need to care about utf-8 collection here?

Reply:
It should be utf8 encoding since we've checked it not `is_binary_like`